### ISSUE TITLE
Function returned by TimeStep networks toFunction() are broken

### ISSUE THREAD

**Issue by lesmo** [BrainJS/brain.js#598]
![giphy (1)](https://user-images.githubusercontent.com/2228196/93001442-922f6a80-f4f4-11ea-8884-ba621413ddc9.gif)

## *What* is wrong?
Function returned by TimeStep networks toFunction() are broken

## *Where* does it happen?
When attempting to use the function returned by `.toFunction()` of a TimeStep network, an undefined variable exception is thrown:

```console
undefined:191
                  product.weights = _input.weights = _this.inputValue;
                                    ^

ReferenceError: _input is not defined
    at eval (eval at toFunction (/.../node_modules/brain.js/dist/brain.js:15256:14), <anonymous>:191:37)
    at Object.<anonymous> (/.../test.js:6:13)
    at Module._compile (internal/modules/cjs/loader.js:774:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:785:10)
    at Module.load (internal/modules/cjs/loader.js:641:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:837:10)
    at internal/main/run_main_module.js:17:11
```

## *How* do we replicate the issue?

1. Train a TimeStep network
2. Run `net.toFunction()` to get a standalone function
3. Try to use returned function
4. ???
5. Profit

```js
const brain = require('brain.js');

const net = new brain.recurrent.LSTMTimeStep();
net.train([[1, 2, 3]]);

const func = net.toFunction();
console.log(func([2, 3, 4]));

```

## Expected behavior (i.e. solution)
Returned function should be functional

## *How* important is this (1-5)?
4... trained time step networks cannot be used without brain.js :(

**Comment by 0J3** [comment]
Can confirm that this issue is still occuring.

Here's some sample code I was working on when running into this issue:
```js
const brain = require('brain.js');
const fs = require('fs');

const net = new brain.recurrent.LSTMTimeStep({
	inputSize: 1,
	hiddenLayers: [1],
	outputSize: 1,
});

const data = [];

const addData = (input, output) => {
	data[data.length] = { input, output };
};

const max = 1;
const diff = 0.4;
const maxNumberLength = 3;

const v = Math.pow(maxNumberLength, 10);
for (let r = 0; r <= max; r = r + diff) {
	for (let g = 0; g <= max; g = g + diff) {
		for (let b = 0; b <= max; b = b + diff) {
			addData(
				{ r, g, b },
				{ bright: Math.floor(((r + g + b) / (max * 3)) * v) / v }
			);
		}
	}
}

net.train(data, { log: true, errorThresh: 0.12 });

fs.writeFileSync(
	'neuralNetworkTrainingData.json',
	JSON.stringify(data, 2, ' ')
);
fs.writeFileSync('neuralNetworkState.json', JSON.stringify(net.toJSON()));
fs.writeFileSync(
	'runNeuralNetwork.js',
	`module.exports=${net.toFunction().toString()}`
);
```
which errors with
![image](https://user-images.githubusercontent.com/67464646/115907432-25585780-a469-11eb-95c1-8259c258a24c.png)
when the function is called

**Comment by robertleeplummerjr** [comment]
This is resolved in version 2.0.0-beta.13, here is a working example: https://observablehq.com/@robertleeplummerjr/brain-js-issue-598

However, this will always be somewhat of an issue with minifiers or obfuscators.
